C#/v1/NoarkWsClientSample/NoarkWsClientSample/Program.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using CommandLine;
+using Documaster.WebApi.Client.Noark5;
+using Documaster.WebApi.Client.Noark5.Client;
+using Documaster.WebApi.Client.Noark5.NoarkEntities;
+using Documaster.WebApi.Client.IDP;
+using Documaster.WebApi.Client.IDP.Oauth2;
+
+namespace NoarkWsClientSample
+{
+    static class Program
+    {
+        public static void Main(string[] args)
+        {
+            var options = ParserCommandLineArguments(args);
+
+            DocumasterClients documasterClients = new DocumasterClients(options);
+
+            SystemInitializationSample initializationSample = new SystemInitializationSample(documasterClients);
+            initializationSample.Execute();
+
+            IntegrationSample integrationSample =
+                new IntegrationSample(documasterClients, options.TestFile1, options.TestFile2);
+            integrationSample.Execute();
+
+            SignOffSample signOffSample = new SignOffSample(documasterClients);
+            signOffSample.SignOffRegistryEntry("56", new List<string>() { "73" });
+
+            FinalizationSample finalizationSample = new FinalizationSample(documasterClients);
```

Comment:
I think we should finalize the structure in the reverse order. That means from the bottom and up.
This will prevent us getting "Cannot edit finalized objects" errors if the user do not have the "edit finalize object" permission.

We had a problem with that on hdir and changing the order fixed it.

Reply:
Fixed. See new sample code.